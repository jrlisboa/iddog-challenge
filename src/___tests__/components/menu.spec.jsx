import React from 'react';
import { shallow } from 'enzyme';
import Menu from '../../components/menu';

import {
    MenuWrapper,
    Item,
} from '../../components/menu/index.style';

describe('Menu component', () => {
    const items = [
        'foo',
        'bar',
        'bazz',
    ];
    const onChangeMock = jest.fn();
    const component = shallow(
        <Menu
            change={onChangeMock}
            activeIndex={0}
            items={items} />,
    );

    it('Should render Menu component with correct content', () => {
        expect(component.find(MenuWrapper)).toBeTruthy();
        expect(component.find(MenuWrapper).children()).toHaveLength(items.length);
    });

    it('Shold call click function items.length times', () => {
        component.find(Item).map((_item) => {
            return _item.simulate('click');
        });
        expect(onChangeMock.mock.calls.length).toBe(3);
    });
});
