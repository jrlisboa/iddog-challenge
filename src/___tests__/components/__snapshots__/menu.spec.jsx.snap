// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component Should match snapshot 1`] = `ShallowWrapper {}`;
